File: prid1920-g13/ClientApp/src/app/models/Discussion.ts
import { Message } from "./Message";
import { User } from "./User";

export class Discussion {
    id: number;
    messages: Message[] = [];
    participants: string [];

    constructor(data) {
        if(data){
            this.messages = data.messages;
            this.participants = data.participants;
            this.id = data.id;
        }
    }
}

File: prid1920-g13/Models/ModelsEntity/Discussion.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using Newtonsoft.Json;
using prid_1819_g13.Models;

namespace prid_1819_g13
{
    public class Discussion
    {
        [Key]
        public int Id {get;set;}
        
        public virtual IList<UserDiscussion> UserDiscussions {get;set;} = new List<UserDiscussion>();
        public virtual IList<Message> Messages {get;set;} = new List<Message>();

        [NotMapped]
        public IEnumerable<User> Participants 
        {
            get => UserDiscussions.Select(u => u.User);
        }
    }
}